### PR TITLE
HARVESTER: additional longhorn information

### PR DESCRIPTION
**Pull request by DaiYuzeng**
#### PR Checklist
- ~Is this a multi-tenancy feature/bug?~
    - [ ] ~Yes, the relevant RBAC changes are at:~
- ~Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?~
    - [ ] ~Yes, the relevant PR is at:~
- ~Are backend engineers aware of UI changes?~
	- [ ] ~Yes, the backend owner is:~

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2856

- We will display more details from longhorn on harvester volume detail page.
- You can see the rebuilding status of volume on the volume list page.

### Areas or cases that should be tested

Rebuilding status on volume page
1. Go to preference page, enable `Extension developer features`
3. Go to create vm with 2 volumes
4. Direct to longhorn volume detail page
5. Go to Support page, open `Embeded Rancher UI` on the new browser tab
6. Go to search type `instance` and then go to instance manager list
7. Delete related instance manager
8. Back to harvester volume page
9. It's expected to see the rebuilding percentage bar

image
image
image
image
image
image
image

More details on volume page

Disk tags / Node tags

1. Go to Edit Node
2. Add Node tags / Disk tags
3. Go to create StorageClass with those tags
4. Go to craete Volume with the StorageClass just created

image

image

image

Last Backup / Last Backup At
1. Go to configure your Backup Target
2. Go to create VM
3. Backup the VM

image

image

### CODE REVIEW

File: pkg/harvester/formatters/HarvesterVolumeState.vue [hunk 1]
```diff
@@ -14,13 +15,49 @@ export default {
...
     warningMessage() {
       return this.row.relatedPV?.metadata?.annotations?.[HCI_ANNOTATIONS.VOLUME_ERROR];
+    },
+
+    rebuildStatus() {
+      return this.row.longhornEngine?.status?.rebuildStatus;
+    },
+
+    isRebuilding() {
+      return this.rebuildStatus && Object.keys(this.rebuildStatus).length > 0;
```
Comment (WuJun2016) [inline pkg/harvester/formatters/HarvesterVolumeState.vue:25]: I think it should be  `>= 0`

File: pkg/harvester/edit/harvesterhci.io.volume.vue [hunk 2]
```diff
@@ -38,9 +38,12 @@ export default {
...
   async fetch() {
     const _hash = {
-      images:    this.$store.dispatch('harvester/findAll', { type: HCI.IMAGE }),
-      snapshots: this.$store.dispatch('harvester/findAll', { type: VOLUME_SNAPSHOT }),
-      storages:  this.$store.dispatch(`harvester/findAll`, { type: STORAGE_CLASS }),
+      images:          this.$store.dispatch('harvester/findAll', { type: HCI.IMAGE }),
+      snapshots:       this.$store.dispatch('harvester/findAll', { type: VOLUME_SNAPSHOT }),
+      storages:        this.$store.dispatch(`harvester/findAll`, { type: STORAGE_CLASS }),
+      longhornVolumes: this.$store.dispatch('harvester/findAll', { type: LONGHORN.VOLUMES }),
+      longhornEngines: this.$store.dispatch('harvester/findAll', { type: LONGHORN.ENGINES }),
```
Comment (WuJun2016) [inline pkg/harvester/edit/harvesterhci.io.volume.vue:45]: Does RBAC have permissions?

Comment (DaiYuzeng) [reply pkg/harvester/edit/harvesterhci.io.volume.vue]: done.

**Review by WuJun2016 — APPROVED** [review]
LGTM

**Comment by WuJun2016** [comment]
@mergify backport release-harvester-v1.2

**Pull request merged**